src/pages/index.js

```diff
@@ -18,96 +19,257 @@ import {
+editAvatarPopup.setEventListeners();
+
+// Likes///
+function handleLikeClick(card) {
+  const isLiked = card.isLiked();
+  const cardID = card._cardID;
+
+  if (isLiked) {
+    api
+      .likeCountRemove(cardID)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error removing like:", error);
+      });
+  } else {
+    api
+      .likeCountAdd(cardID)
+      .then((updatedCard) => {
+        card.setLikes(updatedCard.likes);
+      })
+      .catch((error) => {
+        console.error("Error adding like:", error);
+      });
+  }
+}
+
+const section = new Section({}, containerSelector);
 
 // Functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    (cardID) => handleDeleteClick(card, cardID),
+    handleLikeClick,
+    currentUserId
+  );
+
+  return card.getView();
+}
+
 function addCard(data) {
   const cardData = {
     name: data.title,
     link: data.imageURL,
   };
 
-  renderCard(cardData);
-  addCardFormPopup.close();
+  // Show loading state
+  addNewCardButton.classList.add("showLoading");
+  api
+    .addNewCard(cardData)
+    .then((newCard) => {
+      const cardElement = renderCard(newCard);
+      section.addItem(cardElement);
+      addCardFormPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      addNewCardButton.classList.remove("hideLoading");
+    });
+}
+const api = new Api(apiURL);
+
+const imagePreviewModal = new PopupWithImage(imageModalSelector);
+imagePreviewModal.setEventListeners();
+
+function handleCardImageClick(cardData) {
+  imagePreviewModal.open(cardData);
 }
 
+api
+  .getUserInfo()
+  .then((currentUser) => {
+    currentUserId = currentUser._id;
+
+    api
+      .getInitialCards()
+      .then((cardData) => {
+        cardData.forEach((cardItem) => {
+          const card = new Card(
+            cardItem,
+            "#card-template",
+            handleCardImageClick,
+            handleDeleteClick,
+            handleLikeClick,
+            currentUserId
+          );
+
+          const cardElement = card.getView();
+          section.addItem(cardElement);
+        });
+      })
+      .catch((error) => {
+        console.error("Error fetching initial cards:", error);
+      });
+  })
+  .catch((error) => {
+    console.error("Error fetching user info:", error);
+  });
+
 // card format
 const addCardModalSelector = "#add-card-modal";
-const data = () => {};
 
 const addCardFormPopup = new PopupWithForm(addCardModalSelector, addCard);
 addCardFormPopup.setEventListeners();
-addCardFormPopup.close();
+// addCardFormPopup.close();
 
 addNewCardButton.addEventListener("click", () => {
-  addCardFormValidator.disableButton(); // Disable the submit button
+  addCardFormValidator.disableButton();
 
+  addCardFormPopup.renderLoading(false);
```

Comment:
Needs correction: please remove this line, you should stop the loading after the request in finally block